src/main/java/org/fenixedu/academic/ui/struts/action/administrativeOffice/lists/StudentListByDegreeDA.java

```diff
@@ -196,18 +196,11 @@ public ActionForward searchByDegree(ActionMapping mapping, ActionForm actionForm
             if ((searchBean.isIngressedInChosenYear()) && (registration.getIngressionYear() != executionYear)) {
                 continue;
             }
-
-            if (searchBean.isConcludedInChosenYear()) {
-                Stream<ProgramConclusion> conclusions =
-                        ProgramConclusion.conclusionsFor(registration).filter(ProgramConclusion::isTerminal);
-
-                if (conclusions.allMatch(programConclusion -> {
-                    RegistrationConclusionBean conclusionBean = new RegistrationConclusionBean(registration, programConclusion);
-                    return conclusionBean.getCurriculumGroup() == null || !conclusionBean.isConcluded()
-                            || conclusionBean.getConclusionYear() != executionYear;
-                })) {
-                    continue;
-                }
+            
+            if ((searchBean.hasAnyProgramConclusion())
+            		&& !(ProgramConclusion.conclusionsFor(registration)
+            				.anyMatch(pC -> searchBean.getProgramConclusions().contains(pC)))) {
+            	continue;
```

Comment:
something is missing here ... you should also check if it has a conclusion process with a last version